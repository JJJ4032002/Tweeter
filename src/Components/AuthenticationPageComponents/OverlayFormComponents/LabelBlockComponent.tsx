import React, { useContext } from "react";
import {
  Label,
  LabelBlock,
  SpanText,
  Input,
  ErrorText,
} from "./LabelBlockComponentCss";
import {
  InputStateProps,
  LabelBlockComponentProps,
} from "../../../Interfaces and Types/Types";
import { FormPropsContext } from "../../../Contexts/OverlayFormContext";
function LabelBlockComponent({ children, type }: LabelBlockComponentProps) {
  let {
    Styles,
    FocusAchieved,
    BlurAchieved,
    NameErrRef,
    EmailErrRef,
    InputChange,
    InputStates,
  } = useContext(FormPropsContext);
  let styles = {
    name: "name",
    borderColor: "black",
    txtClr: "",
    fontSize: 1,
    position: 32,
    borderWidth: 1,
  };
  let ref = type === "name" ? NameErrRef : EmailErrRef;
  let SpanName = type === "name" ? "Name" : "Email";
  if (type === "name") {
    styles = Styles[0];
  } else {
    styles = Styles[1];
  }
  return (
    <LabelBlock>
      <Label
        borderWidth={styles.borderWidth}
        borderColor={styles.borderColor}
        htmlFor={type}
      >
        <Input
          value={InputStates[type as keyof InputStateProps]}
          onChange={InputChange}
          name={type}
          onFocus={FocusAchieved}
          onBlur={BlurAchieved}
          type={type}
        />
        <SpanText
          fontSize={styles.fontSize}
          position={styles.position}
          borderColor={styles.borderColor}
        >
          {SpanName}
        </SpanText>
      </Label>
      <ErrorText ref={ref}>{children}</ErrorText>
    </LabelBlock>
  );
}

export default LabelBlockComponent;
