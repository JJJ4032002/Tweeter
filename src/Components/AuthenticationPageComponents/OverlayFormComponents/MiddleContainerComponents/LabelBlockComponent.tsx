import React, { useContext } from "react";
import {
  Label,
  LabelBlock,
  SpanText,
  Input,
  ErrorText,
} from "./LabelBlockComponentCss";
import {
  InputStateProps,
  LabelBlockComponentProps,
} from "../../../../Interfaces and Types/Types";
import { FormPropsContext } from "../../../../Contexts/OverlayFormContext";
function LabelBlockComponent({ children, type }: LabelBlockComponentProps) {
  let {
    Styles,
    FocusAchieved,
    BlurAchieved,
    NameErrRef,
    EmailErrRef,
    InputChange,
    InputStates,
  } = useContext(FormPropsContext);
  let styles = {
    type: "name",
    WhichState: "",
  };
  let ref = type === "name" ? NameErrRef : EmailErrRef;
  let SpanName = type === "name" ? "Name" : "Email";
  if (type === "name") {
    styles = Styles[0];
  } else {
    styles = Styles[1];
  }
  let FinType = "text";
  return (
    <LabelBlock>
      <Label WhichState={styles.WhichState} htmlFor={type}>
        <Input
          value={InputStates[type as keyof InputStateProps]}
          id={type}
          onChange={InputChange}
          name={type}
          onFocus={FocusAchieved}
          onBlur={BlurAchieved}
          type={FinType}
        />
        <SpanText WhichState={styles.WhichState}>{SpanName}</SpanText>
      </Label>
      <ErrorText ref={ref}>{children}</ErrorText>
    </LabelBlock>
  );
}

export default LabelBlockComponent;
