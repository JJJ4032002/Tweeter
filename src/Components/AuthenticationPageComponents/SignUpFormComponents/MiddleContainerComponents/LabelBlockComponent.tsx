import React from "react";
import { LabelBlock } from "./LabelBlockComponentCss";
import { LabelBlockComponentProps } from "../../../../Interfaces and Types/Types";

function LabelBlockComponent({ children }: LabelBlockComponentProps) {
  return <LabelBlock>{children}</LabelBlock>;
}

export default LabelBlockComponent;
