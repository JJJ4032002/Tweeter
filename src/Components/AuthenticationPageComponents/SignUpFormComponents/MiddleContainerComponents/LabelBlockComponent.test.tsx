import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"; // optional
import userEvent from "@testing-library/user-event";
import LabelBlockComponent from "./LabelBlockComponent";

describe("Render label component", () => {
  it("Renders name Label block correctly", () => {
    const { container } = render(
      <LabelBlockComponent type="name">What's your name</LabelBlockComponent>
    );
    expect(container).toMatchSnapshot();
  });
  it("Renders email Label block correctly", () => {
    const { container } = render(
      <LabelBlockComponent type="email">
        Please enter a valid email
      </LabelBlockComponent>
    );
    expect(container).toMatchSnapshot();
  });
  it("Renders password Label block correctly", () => {
    const { container } = render(
      <LabelBlockComponent type="password">
        Please enter a password greater than 8 characters
      </LabelBlockComponent>
    );
    expect(container).toMatchSnapshot();
  });
});
