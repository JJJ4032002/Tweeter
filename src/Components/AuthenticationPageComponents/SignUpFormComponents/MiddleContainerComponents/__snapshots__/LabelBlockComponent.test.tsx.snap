// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render label component Renders email Label block correctly 1`] = `
<div>
  <div
    class="sc-hKwDye iXxlrE"
  >
    <label
      class="sc-jRQBWg cAgqGx"
      for="email"
    >
      <input
        class="sc-iCfMLu XAWHp"
        id="email"
        name="email"
        type="text"
        value=""
      />
      <span
        class="sc-gKclnd chvozb"
      >
        Email
      </span>
    </label>
    <span
      class="sc-dkPtRN jJFEIl"
    >
      Please enter a valid email
    </span>
  </div>
</div>
`;

exports[`Render label component Renders name Label block correctly 1`] = `
<div>
  <div
    class="sc-hKwDye iXxlrE"
  >
    <label
      class="sc-jRQBWg cAgqGx"
      for="name"
    >
      <input
        class="sc-iCfMLu XAWHp"
        id="name"
        name="name"
        type="text"
        value=""
      />
      <span
        class="sc-gKclnd chvozb"
      >
        Name
      </span>
    </label>
    <span
      class="sc-dkPtRN jJFEIl"
    >
      What's your name
    </span>
  </div>
</div>
`;

exports[`Render label component Renders password Label block correctly 1`] = `
<div>
  <div
    class="sc-hKwDye iXxlrE"
  >
    <label
      class="sc-jRQBWg cAgqGx"
      for="password"
    >
      <input
        class="sc-iCfMLu XAWHp"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <span
        class="sc-gKclnd chvozb"
      >
        Password
      </span>
    </label>
    <span
      class="sc-dkPtRN jJFEIl"
    >
      Please enter a password greater than 8 characters
    </span>
  </div>
</div>
`;
